### PR TITLE
retry interrupted system calls

### LINKED ISSUE #679: Retry system calls automatically
Now that Python finally retries system calls automatically (https://www.python.org/dev/peps/pep-0475/) it would be great if the zmq API would behave in the same way and sensibly re-tries interrupted system calls.

### PR DESCRIPTION
**Pull request by minrk**
cc @flub

- cython backend only so far
- retry on EINTR for send,recv,poll,device
- are there other calls that should be retried?

TODO:

- [x] tests
- [x] cffi backend
- [x] determine if there are other calls that might need to be retried (e.g. get/setopt, msg/ctx operations)

closes #679

### CODE REVIEW

**Comment by flub** [comment]
Thanks for getting to this so quickly!

Looking at the manpages getsocketop/setsockopt can raise EINTR as well, it looks like msg/ctx operations are all fine though.

File: zmq/backend/cython/checkrc.pxd [hunk 1]
```diff
@@ -1,7 +1,12 @@
+from libc.stdio cimport printf
```
Comment (flub) [inline zmq/backend/cython/checkrc.pxd]: I've never worked with cython before, but I don't think this is being used?  Probably a left over from some debugging you did?

**Comment by minrk** [comment]
Okay, get/setsockopt are retried now as well, and there are tests for send/recv/poll. Just the CFFI implementation to go.

**Comment by minrk** [comment]
Okay, I think this is good to go, now. Anything more that you can think of?

**Pull request merged**